flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkFactory.java

```diff
@@ -72,6 +70,10 @@ public String factoryIdentifier() {
 
     @Override
     public Set<ConfigOption<?>> requiredOptions() {
-        return ImmutableSet.of(TABLE_NAME, AWS_REGION);
+        final Set<ConfigOption<?>> requiredOptions = new HashSet<>();
+        requiredOptions.add(TABLE_NAME);
+        requiredOptions.add(AWS_REGION);
+
+        return requiredOptions;
```

Comment:
Is this set available for external modification? If yes - `Collections.unmodifiableSet(requiredOptions)` can be used to preserve semantics of immutable collection.

Reply:
This is only used internally for Table API anyway